libs/eavmlib/src/ahttp_client.erl

```diff
@@ -0,0 +1,326 @@
+%
+% This file is part of AtomVM.
+%
+% Copyright 2024 Davide Bettio <davide@uninstall.it>
+%
+% Licensed under the Apache License, Version 2.0 (the "License");
+% you may not use this file except in compliance with the License.
+% You may obtain a copy of the License at
+%
+%    http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+% See the License for the specific language governing permissions and
+% limitations under the License.
+%
+% SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+%
+
+-module(ahttp_client).
+-export([connect/4, request/5, stream/2, stream_request_body/3, recv/2, close/1]).
+
+-define(DEFAULT_WANTED_HEADERS, [<<"Content-Length">>]).
+
+-record(http_client, {host, socket, parser, ref, parse_headers}).
+-record(parser_state, {state, acc, remaining_body_bytes, wanted_headers}).
+
+-type protocol() :: http | https.
+-type host() :: binary() | string().
+-type option() :: gen_tcp:option() | ssl:tls_client_option() | {parse_headers, [binary()]}.
+
+-type backend() :: gen_tcp | ssl.
+
+-type socket_message() :: {tcp, inet:socket(), binary()} | {ssl, ssl:sslsocket(), binary()}.
+
+-type status_response() :: {status, reference(), 0..999}.
+-type header_response() :: {header, reference(), {binary(), binary()}}.
+-type data_response() :: {data, reference(), binary()}.
+-type done_response() :: {done, reference()}.
+
+-type response() :: status_response() | header_response() | data_response() | done_response().
+
+-opaque connection() :: #http_client{}.
+
+-export_type([connection/0]).
+
+-spec connect(Protocol, Host, Port, Options) -> connection() when
+    Protocol :: protocol(),
+    Host :: host(),
+    Port :: inet:port_number(),
+    Options :: [option()].
```

Comment:
I am not sure about all the intricacies of how specs work, this may be perfectly valid, but most of our modules have the format:

`-spec connect(Protocol :: protocol(), Host :: host(), Port :: inet:port_number(), Options :: [option()]) -> Connection :: connection()`


`erlfmt` might use a line split in an entry that long, but that is the pattern that our Erlang modules have been using.

Reply:
Fiexed.